internal/parser/translate.go

```diff
@@ -327,84 +327,155 @@ func fromIngressV1(log logrus.FieldLogger, ingressList []*networkingv1.Ingress)
+func fromHTTPRoutes(log logrus.FieldLogger, httpRouteList []*gatewayv1alpha2.HTTPRoute) ingressRules {
 	result := newIngressRules()
 
 	for _, httproute := range httpRouteList {
 		// first we grab the spec and gather some metdata about the object
-		spec := httproute.Spec
 		objectInfo := util.FromK8sObject(httproute)
+		spec := httproute.Spec
 
-		// next we need to gather all the matching information e.g. paths, methods, e.t.c.
-		methods := make([]*string, 0) // TODO: implement inclusive path/method matches
-		paths := make([]*string, 0)
+		// gather the hostnames that will be used (globally) for route matching
+		hostnames := make([]*string, 0, len(spec.Hostnames))
+		for _, hostname := range spec.Hostnames {
+			hostnames = append(hostnames, kong.String(string(hostname)))
+		}
+
+		// each rule may represent a different set of backend services that will be accepting
+		// traffic, so we make separate routes and Kong services for every present rule.
 		for _, rule := range spec.Rules {
-			for _, match := range rule.Matches {
-				paths = append(paths, match.Path.Value)
+			// the HTTPRoute specification upstream specifically defines matches as
+			// independent (e.g. each match is an OR with other matches, not an AND).
+			// Therefore we treat each match rule as a separate Kong Route, so we iterate through
+			// all matches to determine all the routes that will be needed for the services.
+			routes := make([]kongstate.Route, 0, len(rule.Matches))
+			for matchNumber, match := range rule.Matches {
+				// determine the name of the route, identify it as a route that belongs
+				// to a Kubernetes HTTPRoute object.
+				routeName := kong.String(fmt.Sprintf(
+					"httproute.%s.%s.%d",
+					httproute.Namespace,
+					httproute.Name,
+					matchNumber,
+				))
+
+				// TODO: implement query param matches
+				if len(match.QueryParams) > 0 {
+					log.Errorf("regular expression path matches are not yet supported, HTTPRoute %s/%s can't be routed for match %+v", httproute.Namespace, httproute.Name, match)
+					continue
+				}
+
+				// TODO: implement regex path matches
+				if *match.Path.Type == gatewayv1alpha2.PathMatchRegularExpression {
+					log.Errorf("regular expression path matches are not yet supported, HTTPRoute %s/%s can't be routed for match %+v", httproute.Namespace, httproute.Name, match)
+					continue
+				}
+
+				// build the route object using the method and pathing information
+				r := kongstate.Route{
+					Ingress: objectInfo,
+					Route: kong.Route{
+						Name:         routeName,
+						Protocols:    kong.StringSlice("http", "https"),
+						PreserveHost: kong.Bool(true),
+						Hosts:        hostnames,
+					},
+				}
+				log.Debugf("generated route %s for HTTPRoute %s/%s", routeName, httproute.Namespace, httproute.Name)
+
+				// configuration path matching information about the route if paths
```

Comment:
```suggestion
				// configure path matching information about the route if paths
```

Reply:
changed in a3c46ff307f3d9bd7af25ea0e281c59bbbf83ad8 :+1: